props/saber_BC_buttons.h

```diff
@@ -477,738 +1492,1266 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
+          ToggleSpamBlast();
+        }
+        return true;
+
+#endif  // NUM_BUTTONS == 2 || NUM_BUTTONS == 3
+
+// ----------------------------------------------- Any btn section
+#ifndef BC_DUAL_BLADES
+
+// -------------------- Any btn single blade
+
+// Turn Saber ON
+      case EVENTID(BUTTON_POWER, EVENT_FIRST_SAVED_CLICK_SHORT, MODE_OFF):
+        TurnOnHelper();
+        return true;
+
+// Turn Blade OFF
+      case EVENTID(BUTTON_POWER, EVENT_FIRST_HELD_MEDIUM, MODE_ON):
+        TurnOffHelper();
+        return true;
+
+// Turn Blade ON Muted
+      case EVENTID(BUTTON_NONE, EVENT_TWIST, MODE_OFF | BUTTON_POWER):
+        MuteAll();
+        return true;
+
+// Stab
+      case EVENTID(BUTTON_NONE, EVENT_THRUST, MODE_ON):
+        if (on_pending_ || millis() - saber_on_time_ < 2000) return false; 
+        SaberBase::SetClashStrength(2.0);
+        SaberBase::DoStab();
+        return true;
 
 // Melt
-    case EVENTID(BUTTON_NONE, EVENT_STAB, MODE_ON):
-      //Don't melt if in colorchange mode
-      if (SaberBase::GetColorChangeMode() != SaberBase::COLOR_CHANGE_MODE_NONE) return false;
-      if (SaberBase::IsOn()) { // prevents triggering during preon
+      case EVENTID(BUTTON_NONE, EVENT_STAB, MODE_ON):
+        if (on_pending_) return false;
         clash_impact_millis_ = millis();
         if (!SaberBase::Lockup() && !swinging_) {
           SaberBase::SetLockup(SaberBase::LOCKUP_MELT);
           auto_melt_on_ = true;
           SaberBase::DoBeginLockup();
+          }
+        return true;
+
+#else  // else BC_DUAL_BLADES is defined
+
+// -------------------- Any btn dual blades
+
+// Turn Main Blade ON
+      case EVENTID(BUTTON_POWER, EVENT_FIRST_SAVED_CLICK_SHORT, MODE_OFF):
+        TurnBladeOn(BC_MAIN_BLADE_SET);
+        return true;
+
+// Turn Main Blade ON additionally
+// Blaster Deflection
+      case EVENTID(BUTTON_POWER, EVENT_FIRST_SAVED_CLICK_SHORT, MODE_ON):
+        if (!isMainBladeOn() && !on_pending_) {
+          TurnBladeOn(BC_MAIN_BLADE_SET);
+        } else {
+          DoBlasterBlock();
         }
-      }
-      return true;
+        return true;
 
-// Lightning Block
-    case EVENTID(BUTTON_POWER, EVENT_SECOND_HELD_MEDIUM, MODE_ON):
-      //Don't lightning block if in colorchange mode
-      if (SaberBase::GetColorChangeMode() != SaberBase::COLOR_CHANGE_MODE_NONE) return false;
-      if (SaberBase::IsOn()) { // prevents triggering during preon
-        SaberBase::SetLockup(SaberBase::LOCKUP_LIGHTNING_BLOCK);
-        SaberBase::DoBeginLockup();
+// Turn Main Blade OFF
+      case EVENTID(BUTTON_POWER, EVENT_FIRST_HELD_MEDIUM, MODE_ON):
+        if (isMainBladeOn()) {
+          TurnBladeOff(BC_MAIN_BLADE_SET);
+        }
+        return true;
+
+// Thrust ON either non-ignited blade
+// Stab
+      case EVENTID(BUTTON_NONE, EVENT_THRUST, MODE_ON):
+        if (millis() - last_thrust_millis_ < 200 ||
+            millis() - saber_on_time_ < 2000) return false;        
```

Comment:
move return to next line to make it easier to see

Reply:
done